libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -267,8 +268,8 @@ void AP_Mission::reset()
 ///     returns true if mission was running so it could not be cleared
 bool AP_Mission::clear()
 {
-    // do not allow clearing the mission while it is running
-    if (_flags.state == MISSION_RUNNING) {
+    // do not allow clearing the mission while it is running unless disarmed
+    if (AP::arming().is_armed() && _flags.state == MISSION_RUNNING) {
```

Comment:
I think normally we used the AP_HAL::is_soft_armed() (or something like that) to determine if the vehicle is armed or not.  We have some duplicate state and while they may be same, there's always a chance that they're not so let's stick to the most commonly used method.